sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/healthcare/HL7v2IO.java

```diff
@@ -466,21 +475,63 @@ private Message fetchMessage(HealthcareApiClient client, String msgId)
       this.hl7v2Stores = hl7v2Stores;
       this.filter = filter;
       this.initialSplitDuration = null;
+      this.timestampMethod = DEFAULT_TIMESTAMP_METHOD;
     }
 
-    public ListHL7v2Messages withInitialSplitDuration(Duration initialSplitDuration) {
+    /**
+     * Controls the initial splitting for parallelization of HL7v2 Messages List requests based on
+     * disjoint sendTime filters of the specified duration.
+     *
+     * @param initialSplitDuration the initial split duration
+     * @return the list hl 7 v 2 messages
+     */
+    ListHL7v2Messages withInitialSplitDuration(Duration initialSplitDuration) {
       this.initialSplitDuration = initialSplitDuration;
       return this;
     }
 
+    /**
+     * Controls if the output elements will be assigned a timestamp beased on the sendTime property
+     * or the input element's timestamp.
+     *
+     * @param timestampMethod the timestamp method
+     * @return the list hl 7 v 2 messages
+     */
+    ListHL7v2Messages withTimestampMethod(TimestampMethod timestampMethod) {
```

Comment:
How about `withOutputTimestampMethod`, also check that the argument is non-null?

nit: I'm torn between using an enum or just having the two variants between methods. Your call. Having two variants prevents the user from passing in null here.